### DISCUSSION TITLE
Hi! Newbie with API v2, need help to understand how to connect and retrieve data

### DISCUSSION THREAD

**Discussion by christianmilano** [fattureincloud/api#254] [Q&A] [answered]
Hi, I was ok using v1 of API, but now I'm not understanding what I need and what I really need to do to have the same result.
I've created a new Application and get the Client ID, also I have connected the new Application and get the Token.

somebody can please explain how to "translate" something like this v1 to new v2?:

Thanks!

```php
$url = "https://api.fattureincloud.it/v1/fatture/lista";

$request = array("api_uid" => "******", "api_key" => "******************************, "data_inizio" => "$data_inizio", "data_fine" => "$data_fine");

$options = array( 
     "http" => array(
          "header"  => "Content-type: text/json\r\n",
          "method"  => "POST",
          "content" => json_encode($request)
     ),
);

$context  = stream_context_create($options);
$result = json_decode(file_get_contents($url, false, $context), true);
print '<pre>';    
print_r($result);
print '</pre>'; 

for($i = 0; $i < sizeof($result["lista_documenti"]); $i++) {
     echo  $id = $result["lista_documenti"][$i]["id"];
     echo $id_cliente = $result["lista_documenti"][$i]["id_cliente"];
     echo $nome = $result["lista_documenti"][$i]["nome"];
}
```

**Comment by valmoz** [comment, marked as answer]
Hi @christianmilano 
we prepared a [Migration Guide](https://developers.fattureincloud.it/docs/api-v2-migration) to help you with this task.

You basically need to:

- [Create an app](https://developers.fattureincloud.it/docs/basics/create-an-app)
- [Select an authentication method](https://developers.fattureincloud.it/docs/basics/authentication)
- Obtain a token using the selected method
- Perform the API call

The method you need to use is the [List Issued Documents](https://developers.fattureincloud.it/api-reference#get-/c/-company_id-/issued_documents).
To help speed up the development process, we suggest you use the [PHP SDK](https://developers.fattureincloud.it/docs/sdks/php-sdk).

You can also find a [quickstart](https://developers.fattureincloud.it/docs/quickstarts/php-sdk-quickstart) implementing a List method. It uses the List Suppliers method, but replacing it with the List Issued Documents method should be easy.

Let us know if you need further help.
Thanks
Mauro

**Reply by valmoz** [reply]
Hi @christianmilano can we close this discussion?
Thanks
Mauro

**Comment by christianmilano** [comment]
Yes thanks

Christian Serra
***@***.***
——————————————————————————————————————————
CHRISTIAN SERRA
20123 Milano Italy - Via Gioberti 2
www.christianserra.com
Tel. +39 3498780671
..............................................................................................................
The contents of this e-mail and any attachments are confidential to the intended recipient. 
If this email is received in error, please immediately notify the sender and delete the e-mail and attachments. 
Please note that neither the sender nor the sender’s company accept any responsibility for viruses 
and it is your responsibility to scan or otherwise check this e-mail and any attachments.
